### DISCUSSION TITLE
Purpose of `deadline` in 5-t-swap-audit

### DISCUSSION THREAD

**Discussion by trashpirate** [Cyfrin/security-and-auditing-full-course-s23#249] [Q&A] [answered]
Could someone explain more in detail what the reasoning is behind having a `deadline` for the  deposit and swap functions in `TSwapPool`? I know that this is common practice on DEXs but I don't quite understand why it is needed. Appreciate any insights.

**Comment by EngrPips** [comment, marked as answer]
Hello, in a Decentralized exchange, price is controlled by demand and supply, and as time goes on, the demand and supply drive the price of the tokens in the pool. So if you wanted to swap 1 token A and based on the current price of the pool, you are supposed to get 5 token B. If you don't have the `deadline` parameter and for on reason or the other your trade got delayed till some later time in the future you might end up getting 2 token B for your 1 token A or even less. The same is the situation for `deposit` too but your shares of the pool LP is affected in that scenario. Please ask question if anything is confusing

**Reply by trashpirate** [reply]
That makes sense, thank you! But isn't the same possible if many transactions are submitted within the same block? Wouldn't the slippage parameter prevent the scenario you are describing?

**Reply by EngrPips** [reply]
Yeah, Looking at it like this would make the `deadline` parameter redundant, but look at it from a perspective where you want to swap to take a time-boxed advantage on another Defi protocol. In this case, if the opportunity you wish to utilize on the other Defi protocol is no more, then you don't need to make the swap anymore then you can really see the essence of the `deadline` parameter

**Reply by trashpirate** [reply]
Ah, got it! Thanks for the detailed response!

**Reply by EngrPips** [reply]
Anytime, Friend.
